server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -171,14 +172,46 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
 		return nil, err
 	}
 
-	if err := userDB.AddUserToGroup(ctx, groupID, jwtUser.UserID); err != nil {
+	if err := userDB.AddUserToGroup(ctx, jwtUser.UserID, groupID); err != nil {
 		return nil, err
 	}
 	return &grpb.CreateGroupResponse{
 		Id: groupID,
 	}, nil
 }
 
+func (s *BuildBuddyServer) JoinGroup(ctx context.Context, req *grpb.JoinGroupRequest) (*grpb.JoinGroupResponse, error) {
+	auth := s.env.GetAuthenticator()
+	userDB := s.env.GetUserDB()
+	if auth == nil || userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	user, err := userDB.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	group := &tables.Group{
+		GroupID: req.GetId(),
+	}
+	if err := userDB.FillGroup(ctx, group); err != nil {
+		return nil, err
+	}
+	// If the user's email matches the group's owned domain, they can be added
+	// as a member immediately.
+	if group.OwnedDomain != "" && GetEmailDomain(user.Email) == group.OwnedDomain {
```

Comment:
might be slightly more readable if you swap the order of these two: GetEmailDomain(user.Email) == group.OwnedDomain

then you have if x != nil && x.y == z {}

Reply:
Done